### ISSUE TITLE
Randomly cannot disconnect from AP

### ISSUE THREAD

**Issue by murphyalexandre** [spark/spark-setup-android#14]
Sometimes when using the android app it won't disconnect from the photon's AP and I have to do a lot of resetting, clearing the wifi, even reflashing the firmware for it to work!

However, this is where things get interesting, I have no issue whatsoever with the Particle app from the Play Store. Go figure.

Is there something new in the official app that wasn't put in this example app ?

**Comment by jensck** [comment]
Strange!  The app in the store doesn't have any special sauce -- the setup is 100% powered by the publically available SDKs.   The one thing that is different about the app in the store is that it's using a slightly out of date version of the setup lib and SDKs.

Can you try to reproduce using cloud SDK v0.2.2 and device setup v0.3.1 ?  Those are the versions used in the current Play Store release.

**Comment by murphyalexandre** [comment]
You know what, I was using a Samsung Galaxy Note 4 when the bug keep on happening and since then, I switched to my Nexus 6P and the problem is gone. 

I've read somewhere in the forums that people were having troubles with Samsung devices so I'm guessing it's an issue specific to those devices and there must have a bug within the wifi driver or something.

Closing this issue for the time being.
